tests/unittests/core/fileio/fileutilstest.cpp

```diff
@@ -0,0 +1,320 @@
+  FilePath rootFileHidden{root.getPathTo(".hidden.txt")};
+
+  FilePath subdir{root.getPathTo("subdir")};
+  FilePath subdirFile{subdir.getPathTo("file.txt")};
+  FilePath subdirSubdir{subdir.getPathTo("subdir")};
+  FilePath subdirSubdirFile{subdirSubdir.getPathTo("file.txt")};
+  FilePath subdirSubdirFileHidden{subdirSubdir.getPathTo(".hidden.txt")};
+
+  // the destinations for copying files, nonexistent at start of test
+  FilePath rootFileMissing{root.getPathTo("missing.txt")};
+  FilePath rootFileCopy{root.getPathTo("fileCopy.txt")};
+  FilePath subdirCopy{root.getPathTo("subdirCopy")};
+  FilePath subdirCopyFile{subdirCopy.getPathTo("file.txt")};
+  FilePath subdirCopySubdir{subdirCopy.getPathTo("subdir")};
+  FilePath subdirCopySubdirFile{subdirCopySubdir.getPathTo("file.txt")};
+  FilePath subdirCopySubdirFileHidden{
+      subdirCopySubdir.getPathTo(".hidden.txt")};
+
+  QStringList filter{"*.txt"};
+
+  FileUtilsTest() {
+    QDir().mkdir(root.toNative());
+    QDir().mkdir(subdir.toNative());
+    QDir().mkdir(subdirSubdir.toNative());
+
+    setupFile(rootFile, "test\n");
+    setupFile(rootFileHidden, "hiddenContent\n", true);
+    setupFile(subdirFile, "test\n");
+    setupFile(subdirSubdirFile, "test\n");
+    setupFile(subdirSubdirFileHidden, "hiddenContent\n", true);
+  }
+
+  ~FileUtilsTest() {
+    // each test should clean itself after run
+    QDir(root.toNative()).removeRecursively();
+  }
+
+  /**
+   * @brief Make the unordered result list comparable with better test message
+   *
+   *  **using std::string:**
+   *  Expected equality of these values:
+   *    comparable(actual)
+   *      Which is: "\n/tmp/librepcb/1696569432486_2075782095/file.txt..."
+   *    comparable(expected)
+   *      Which is: "\n/tmp/librepcb/1696569432486_2075782095/file.txt..."
+   *  With diff:
+   *   @@ +1,4 @@
+   *
+   *   +/tmp/librepcb/1696569432486_2075782095/file.txt
+   *    /tmp/librepcb/1696569432486_2075782095/file.txt
+   *    /tmp/librepcb/1696569432486_2075782095/subdir/file.txt
+   *
+   *  **using Qt's QString**:
+   *  Expected equality of these values:
+   *    comparable(actual)
+   *      Which is: { 2-byte object <0A-00>, 2-byte object <2F-00>, ... }
+   *    comparable(expected)
+   *      Which is: { 2-byte object <0A-00>, 2-byte object <2F-00>, ... }
+   *
+   *  @param path result from getFilesInDirectory
+   *  @return comparable string for EXPECT macros
+   */
+  static std::string comparable(const QList<FilePath>& path) {
+    auto join = [](std::string a, std::string b) {
+      return !a.empty() ? (std::move(a) + '\n' + std::move(b)) : std::move(b);
+    };
+    auto conv = [](const FilePath& filepath) {
+      return filepath.toStr().toStdString();
+    };
+    std::vector<std::string> paths{};
+    std::transform(path.begin(), path.end(), std::back_inserter(paths), conv);
+    std::sort(paths.begin(), paths.end());
+    return std::accumulate(paths.begin(), paths.end(), std::string(), join);
```

Comment:
Huh this function looks very complicated for the simple thing it does :see_no_evil: I'm not even sure if the `std::move()` calls actually make sense in this context.

With Qt it would be not only much easier to understand, but even shorter:

```
QStringList pathsStr;
foreach (const FilePath& fp, paths) {
  pathsStr.append(fp.toStr());
}
pathsStr.sort();
return pathsStr.join("\n").toStdString();
```
Anyway, your solution seems to work so this time I could live with that...

Reply:
I forgot that Qt has "join" function that is missing in plain c++, and I agree that using it produces more readable code. 

But this code part you recommended is Yours, during squashing, your authorship will be deleted. I mean it will be better that you will push your commit rewriting this function content just after squash merge of this PR. Both versions will be in the history for comparison.